lib/install/validate/validator.go

```diff
@@ -127,20 +118,60 @@ func CreateNoDCCheck(ctx context.Context, input *data.Data) (*Validator, error)
 
 	v.Session.Populate(ctx)
 
+	// only allow the datacenter to be specified in the taget url, if any
+	pElems := strings.Split(v.DatacenterPath, "/")
+	if len(pElems) > 2 {
+		detail := "--target should only specify datacenter in the path (e.g. https://addr/datacenter) - specify cluster, resource pool, or folder with --compute-resource"
+		log.Error(detail)
+		v.suggestDatacenter()
+		return nil, errors.New(detail)
+	}
+
 	return v, nil
 }
 
 func (v *Validator) datacenter() error {
 	if v.Session.Datacenter == nil {
-		detail := "Target should specify datacenter when there are multiple possibilities, e.g. https://addr/datacenter"
+		detail := "Datacenter must be specified in --target (e.g. https://addr/datacenter)"
 		log.Error(detail)
-		// TODO: list available datacenters
+		v.suggestDatacenter()
 		return errors.New(detail)
 	}
 	v.DatacenterPath = v.Session.Datacenter.InventoryPath
 	return nil
 }
 
+// suggestDatacenter suggests all datacenters on the target
+func (v *Validator) suggestDatacenter() {
+	defer trace.End(trace.Begin(""))
+
+	log.Info("Suggesting valid values for datacenter in --target")
+
+	dcs, err := v.Session.Finder.DatacenterList(v.Context, "*")
+	if err != nil {
+		log.Errorf("Unable to list datacenters: %s", err)
+		return
+	}
+
+	if len(dcs) == 0 {
+		log.Info("No datacenters found")
```

Comment:
I'm going to leave this as it's consistent with the other suggestions and we didn't fail to query the environment just got no results